code/datums/atom_hud.dm

```diff
@@ -101,6 +103,8 @@ GLOBAL_LIST_INIT(huds, list(
 	if(!A || (A in hudatoms))
 		return FALSE
 	hudatoms |= A
+	if(istype(A, /obj/effect/temp_visual))
+		RegisterSignal(A, COMSIG_PARENT_QDELETING, .proc/remove_from_hud)
```

Comment:
A) Why only temp visuals and not all atoms
B) I'm  think this registers twice in some cases because of protections we added recently